src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/InputAbbreviationAsWordInNameCheckEnhancedInstanceof.java

```diff
@@ -3,25 +3,29 @@
 
 import java.util.ArrayList;
 import java.util.Locale;
+// config:
+// allowedAbbreviationLength="1"
 
 public class InputAbbreviationAsWordInNameCheckEnhancedInstanceof {
 
     public void t(Object o1, Object o2) {
         // Should cause two violations, STRING and INTEGER, capital letter count > allowed
-        if (!(o1 instanceof String STRING) && (o2 instanceof Integer INTEGER)) {}
+        if (!(o1 instanceof String STRING) // violation
+                && (o2 instanceof Integer INTEGER)) {} // violation
 
         ArrayList<Integer> arrayList = new ArrayList<Integer>();
-        if (arrayList instanceof ArrayList<Integer> aXML) { // ok when allowed "XML" abbreviation
+        if (arrayList instanceof ArrayList<Integer> aXML) { // ok
             System.out.println("Blah");
         }
 
         boolean result = (o1 instanceof String a1) ?
-                (o1 instanceof String aTXT) :   // violation, capital letter count > allowed
+                (o1 instanceof String aTXT) :   // ok
```

Comment:
why no violation ?

Reply:
It looks like this file is now being checked with the default configuration with allowedAbbreviationLength=3

Reply:
All comments should be correct now.